### PR TITLE
Migrate Colony Creation Wizard to use React Hook Forms

### LINKED ISSUE #116: Migrate Colony Creation Wizard to use React Hook Form
See #103 -> same idea but for colony creation wizard

### PR DESCRIPTION
**Pull request by willm30**
## Description

This PR migrates the Colony Creation wizard from Formik to React Hook Form. It should work the same as before, just now it's using the React Hook Form components.

**New stuff** ✨

* HookForm-specific `InputStatus`
* `utils/network`: some helpers to see what network speed the user has. May be useful to conditionally show a loading status based on this information. Faster connections may make a loading status redundant, e.g. if we're querying our own db, it's typically very fast and thus no need to display a loading status. The loading status also appears to flash in this case, because of the connection speed, which is also good to avoid.

**Changes** 🏗

* `ColonyCreationWizard`: now using React Hook Form components.
* Tidied up imports in `UserCreationWizard`

Resolves  #116

### CODE REVIEW

File: src/components/common/CreateColonyWizard/TokenSelector.tsx [hunk 1]
```diff
@@ -83,92 +60,54 @@ const getStatusText = ({
...
 
+const getTokenData = (tokenQuery?: GetTokenByAddressQuery) => {
+  const tokenData = tokenQuery?.getTokenByAddress?.items;
+  const { name, symbol } = tokenData?.[0] || {};
+  return { name, symbol };
+};
+
+const getLoadingState = (isLoading: boolean) =>
+  // Don't show loading state if connection is 4G or unknown
+  connectionIs4G() !== false ? false : isLoading;
```
Comment (willm30) [inline src/components/common/CreateColonyWizard/TokenSelector.tsx:81]: The thought here is that on quick connections, the loading state is so brief that it looks better not to show it (else it flashes for a microsecond and then disappears).

File: src/components/common/CreateColonyWizard/TokenSelector.tsx [hunk 2]
```diff
@@ -83,92 +60,54 @@ const getStatusText = ({
...
-  ] = useGetTokenByAddressLazyQuery();
-
-  const tokenData = tokenQuery?.getTokenByAddress?.items;
-  const { name, symbol } = tokenData?.[0] || {};
-  const prevTokenAddress = usePrevious(tokenAddress);
+    skip: !isValid,
+    onCompleted: handleComplete,
+  });
 
-  const fetchToken = (event: React.ChangeEvent<any>) => {
```
Comment (willm30) [inline src/components/common/CreateColonyWizard/TokenSelector.tsx:124]: Moved to validation schema

File: src/components/shared/Fields/InputStatus/HookForm/InputStatus.tsx [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+  const Element = appearance.direction === 'horizontal' ? 'span' : 'p';
+  return (
+    <Element
+      className={getMainClasses(appearance, styles, {
+        error: !!error && !isLoading,
+        hidden:
+          (!text && !isLoading) || (!!error && !isNil(touched) && !touched),
+      })}
+    >
+      {isLoading ? loadingText : text}
```
Comment (willm30) [inline src/components/shared/Fields/InputStatus/HookForm/InputStatus.tsx:52]: Bringing the loading state into this component means we don't have to think about it higher up the component hierarchy, we can just pass in an `isLoading` prop.

**Review by rdig — APPROVED** [review]
Nothing sticks out and follows the same lines as you did for the user creation wizard refactor.

Nice work.

File: src/components/shared/Fields/Input/HookForm/InputComponent.tsx [hunk 4]
```diff
@@ -22,6 +22,7 @@ const LengthWidget = ({ length, maxLength }: LengthWidgetProps) => (
 export interface HookFormInputComponentProps
   extends Omit<InputProps, 'placeholder'> {
   placeholder?: string;
+  inputValueLength: number;
 }
 
 const displayName = 'HookFormInputComponent';
```
Comment (Julianahlk) [inline src/components/shared/Fields/Input/HookForm/InputComponent.tsx:28]: This display name could be fixed.

File: src/components/shared/Fields/InputStatus/HookForm/InputStatus.tsx [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+    'appearance' | 'status' | 'statusValues' | 'isLoading' | 'loadingAnnotation'
+  > {
+  /** Error text (if applicable) */
+  error?: Message;
+
+  /** Has input field been touched? */
+  touched?: boolean;
+}
+
+const displayName = 'HookFormInputStatus';
```
Comment (Julianahlk) [inline src/components/shared/Fields/InputStatus/HookForm/InputStatus.tsx:24]: This display name could be fixed.

**Review by Julianahlk — APPROVED** [review]
All good, nice work! Just left comments about the display names.

**Review by chinins — APPROVED** [review]
Great job!

**Pull request merged**